### PR TITLE
Edit folder structure and add setuptools support

### PR DESCRIPTION
**Pull request by albertosottile**
As discussed in issue #207, our current folder structure is different from the typical one for a Python package. Specifically, the `resources` folder should be included in the `syncplay` folder, as all the relevant python script.

This PR includes that file restructuring and also adds a `setup.py` script that relies on it. Entry points are used for both client and server. Requirements are taken directly from the corresponding text files (thanks to PR #230). 

Additionally, the packages needed for TLS are now included in the main requirements, as we do not really want the TLS support to be optional. Also, AppVeyor has been updated to the latest `py2exe` ([link](https://github.com/albertosottile/py2exe/releases/tag/v0.9.3.0)) to enable the embedding of TLS libraries. Incidentally, Windows binaries are also upgraded to Python 3.6.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import setuptools
+
+from syncplay import projectURL, version as syncplay_version
+
+def read(fname):
+    with open(fname, 'r') as f:
+        return f.read()
+
+installRequirements = read('requirements.txt').splitlines() +\
+                      read('requirements_gui.txt').splitlines()
```
Comment (ObserverOfTime) [inline setup.py]:
```suggestion
```

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import distutils.command.install_scripts
+import setuptools
+
+from syncplay import projectURL, version as syncplay_version
+
+def read(fname):
+    with open(fname, 'r') as f:
+        return f.read()
+
+installRequirements = read('requirements.txt').splitlines() +\
```
Comment (ObserverOfTime) [inline setup.py]:
```suggestion
```

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    description=' '.join([
+        'Client/server to synchronize media playback',
+        'on mpv/VLC/MPC-HC/MPC-BE on many computers'
+    ]),
+    long_description=read('README.md'),
+    long_description_content_type="text/markdown",
+    url=projectURL,
+    download_url=projectURL + 'download/',
+    packages=setuptools.find_packages(),
+    install_requires=installRequirements,
```
Comment (ObserverOfTime) [inline setup.py]:
```suggestion
    install_requires=read('requirements.txt').splitlines(),
    extras_require={'gui': read('requirements_gui.txt').splitlines(),
```

**Comment by Et0h** [comment]
> It might be better to place `requirements_gui.txt` in `extras_require` since some users may not want the GUI (and `PySide2` is several MBs large). You'll just need to inform end users that they'll need `syncplay[gui]` for the full package.

Aside from saving a few megabytes of storage space and speeding up the installation time, are there any circumstances where automatically installing PySide2 would be disruptive, prohibitive or impossible?

**Comment by Et0h** [comment]
@ObserverOfTime
> It might be better to place `requirements_gui.txt` in `extras_require` since some users may not want the GUI (and `PySide2` is several MBs large). You'll just need to inform end users that they'll need `syncplay[gui]` for the full package.

Aside from saving a few megabytes of storage space and speeding up the installation time, are there any circumstances where automatically installing PySide2 would be disruptive, prohibitive or impossible?

**Comment by ObserverOfTime** [comment]
I think #229 qualifies as a use case where the GUI is not needed.

**Comment by daniel-123** [comment]
Personally I think that the vast majority of users do use the GUI and would expect the "software_name" package to deliver that functionality. I would prefer not to even slightly inconvenience them by making them install package with extras. Even if there exist some use cases where such packaging would be more convenient.

What is possibly worth considering is that most cases where one would care about not including superfluous dependencies are headless server. So maybe that should be a completely separate, minimalist package?

I'm not entirely familiar with setuptools ecosystem, but at least with deb/rpm packaging such problem would be usually solved by existence of package syncplay-no-gui that's either required by full syncplay or is mutually exclusive with it.

**Comment by ObserverOfTime** [comment]
I don't think what you're proposing is feasible with `setuptools`.
Additionally, it's not possible to declare default or opt-out requirements.

How about this solution:
```diff
diff --git a/setup.py b/setup.py
index 91b2602..a083b15 100644
--- a/setup.py
+++ b/setup.py
@@ -3,14 +3,16 @@
 import distutils.command.install_scripts
 import setuptools
 
+from os import environ as env
 from syncplay import projectURL, version as syncplay_version
 
 def read(fname):
     with open(fname, 'r') as f:
         return f.read()
 
-installRequirements = read('requirements.txt').splitlines() +\
-                      read('requirements_gui.txt').splitlines()
+installRequirements = read('requirements.txt').splitlines()
+if env.get('SYNCPLAY_NO_GUI', '0') != '1':
+    installRequirements += read('requirements_gui.txt').splitlines()
 
 setuptools.setup(
     name="syncplay",
```

This way, regular users will not be affected and anyone who doesn't want the GUI can just set `SYNCPLAY_NO_GUI=1` prior to the installation and update.

**Comment by albertosottile** [comment]
Will that work for wheels too? I wonder if the sake of not having `PySide2` installed on a system is worth amending in this way the `setup.py` script, introducing a new environment variable, documenting it somewhere, maintaining the whole thing...

**Comment by Et0h** [comment]
@ObserverOfTime Thanks for your suggestion regarding how we could use environmental variables as a way of supporting a 'no GUI required' installation option without making it the default - we will look into it further if we ever decide to pursue PyPi support.

**Pull request merged**